pkg/apis/experimental/v1alpha1/stack_types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+)
+
+const StackKind = "Stack"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object,k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMetaAccessor
+
+type Stack struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StackSpec   `json:"spec"`
+	Status StackStatus `json:"status"`
+}
+
+type StackSpec struct {
+	Id         string     `json:"id"`
+	BuildImage StackImage `json:"buildImage"`
+	RunImage   StackImage `json:"runImage"`
+}
+
+type StackImage struct {
+	Image string `json:"image"`
+}
+
+type StackStatus struct {
+	duckv1alpha1.Status `json:",inline"`
+
+	BuildImageRef string `json:"buildImageRef"`
+	RunImageRef   string `json:"runImageRef"`
```

Comment:
I am concerned that ref in k8s might imply another resource. On our builders we just call the resolved RunImage `status.stack.runImage`. However, for consistency, I wonder if we should have `status.runImage.latestImage` here and on our builder status.

Reply:
Makes sense.